### ISSUE TITLE
New logo for pygame ce

### ISSUE THREAD

**Issue by gresm** [pygame-community/pygame-ce#2234]
Any thoughts about creating a new logo for pygame-ce? In my opinion, it doesn't have to be done from scratch, an edited version of a logo would be also fine. This change would be made for better distinction between pygame and pygame-ce and would likely also affect an icon that is used as default if no icon is provided by the program.

**Comment by yolowex** [comment]
I do think that we should change the logo, but after a long-exhausting-fruitful conversation in the discord server, 
we agreed upon a mutual idea that we should wait until pygame-ce crosses its infantile age. ( and also the fact that pygame-ce is pygame, not something that's trying to coexist and compete with it )

with all that in mind, maybe six month later, or even a year later, the steering council and the contributors will be willing to change or re-design the logo.

but I'm not sure.

**Comment by yolowex** [comment]
![pygame-badge-SMA-3051624370](https://github.com/pygame-community/pygame-ce/assets/122750743/7e493b0f-b6f0-4cdc-aa19-c295e799c5ff)
![images](https://github.com/pygame-community/pygame-ce/assets/122750743/47e15740-4b7e-4e4e-bed4-ec561db69f08)

these are some logo's I found in internet, I don't know who made these, or if they are in public domain, but the first one seems really good ( for an android app maybe? )

I'm thinking of these two more of optional designs rather than a official logo

**Comment by novialriptide** [comment]
I personally don't like either.

**Comment by gresm** [comment]
IMO the first one is more suitable as a splash screen (but considering that you don't know the images licenses, I think that those two are currently out of consideration).

**Comment by yolowex** [comment]
I told you people think we're a bunch of kids :/ 
![adults](https://github.com/pygame-community/pygame-ce/assets/122750743/44ad5025-8292-49b8-8a38-de4434c8abb4)

**Comment by hakkeroid** [comment]
Hello, I just discovered the community edition of pygame and somehow got intrigued by this topic. I fully understand you are not looking for a new logo just yet. However, in case you ever feel like changing it, I will throw my sketch into the basket.
![logo](https://github.com/pygame-community/pygame-ce/assets/10213186/399f5e1d-9843-493a-a4ff-12f9aeb4482c)

**Comment by yunline** [comment]
@hakkeroid I like your design. 
However, the color looks like the icon of Arduino
![image](https://github.com/pygame-community/pygame-ce/assets/31395137/8d82e496-e24e-4132-b1c7-5433f253c9ce)
So I tried to make another version based on yours.
![pygame](https://github.com/pygame-community/pygame-ce/assets/31395137/dea0d88e-b5fd-43e9-9eb2-9e48c7689f33)
I hope this would help you.

**Comment by yolowex** [comment]
> @hakkeroid I like your design. 
> However, the color looks like the icon of Arduino
> ![image](https://github.com/pygame-community/pygame-ce/assets/31395137/8d82e496-e24e-4132-b1c7-5433f253c9ce)
> So I tried to make another version based on yours.
> ![pygame](https://github.com/pygame-community/pygame-ce/assets/31395137/dea0d88e-b5fd-43e9-9eb2-9e48c7689f33)
> I hope this would help you.
> 

I like this, but isn't the convention of pygame to write it fully in lowercase?
PyGame gives me weird vibes

**Comment by Conz3D** [comment]
After some discussion under off-topic on discord, I came to the conclusion, that seperating the logo from the snake (snek) would be the best way to go. 
It's more of a definition to guide the discussion than a design draft:
- the pygame balloony text is _"The Logo"_
- No snake in the logo, but "snek" as _"The Mascot"_.

It would be close to the original pygame "logo" to not provocate in the current pygame situation. But would give us enough space to "play" with the snake.
Like in corp. logo design, where a logo is changed only slighly in most cases of a re-design.

For my liking is the "pygame" text a little bit too ballooney, But the redesigned one should follow the current style, slighly tuned.
Text "pygame-ce" designed in monochrome as _"The Logo"_. 
The "pygame-ce" is a little bit long and "broken" with the "-ce". Maybe there is a way to get rid of the "-" in the logo.

On the API Docs pages, I would only use the pygame-ce text logo, without the mascot "snek".
On other pages it can be the text logo with snek (with the option to pose snek depending on the content of the page, like for FAQ: "Snake is looking to the side with a questionmark over his/her head".
On the landing page a little bit bigger fancy version of the logo with snek. 
But the official logo is only the tuned balloony text. Maybe this text logo on every page as home button.

I hope this will help the discussion in some way to find a middle ground.

**Comment by yolowex** [comment]
> After some discussion under off-topic on discord, I came to the conclusion, that seperating the logo from the snake (snek) would be the best way to go. It's more of a definition to guide the discussion than a design draft:
> 
>     * the pygame balloony text is _"The Logo"_
> 
>     * No snake in the logo, but "snek" as _"The Mascot"_.
> 
> 
> It would be close to the original pygame "logo" to not provocate in the current pygame situation. But would give us enough space to "play" with the snake. Like in corp. logo design, where a logo is changed only slighly in most cases of a re-design.
> 
> For my liking is the "pygame" text a little bit too ballooney, But the redesigned one should follow the current style, slighly tuned. Text "pygame-ce" designed in monochrome as _"The Logo"_. The "pygame-ce" is a little bit long and "broken" with the "-ce". Maybe there is a way to get rid of the "-" in the logo.
> 
> On the API Docs pages, I would only use the pygame-ce text logo, without the mascot "snek". On other pages it can be the text logo with snek (with the option to pose snek depending on the content of the page, like for FAQ: "Snake is looking to the side with a questionmark over his/her head". On the landing page a little bit bigger fancy version of the logo with snek. But the official logo is only the tuned balloony text. Maybe this text logo on every page as home button.
> 
> I hope this will help the discussion in some way to find a middle ground.

What about not including the ce part at all, because pygame-ce *is* pygame

**Comment by novialriptide** [comment]
@yolowex 
people won't be able to distinguish the difference between the two then.

**Comment by yolowex** [comment]
> @yolowex people won't be able to distinguish the difference between the two then.

I'm more concerned about the possible legal problems. wouldn't be against some rules if a project uses the logo of another project?
ofc pygame-ce is rooted in pygame, but now, they are different projects, with different maintainers and licenses.
so it's only rational to be worried about such a thing.

**Comment by bilhox** [comment]
Hi,

In my opinion, the only change the logo should have, is adding ``ce`` at the bottom right so we can know it's pygame-ce.
This is really useful, especially in the docs like in my website design, I don't like the idea to set the title to ``pygame-ce - documentation`` and would like somehow set it to just ``documentation``.

What do you think ? We can just have this logo only for documentation if you want.

**Comment by yolowex** [comment]
> Hi,
> 
> In my opinion, the only change the logo should have, is adding `ce` at the bottom right so we can know it's pygame-ce. This is really useful, especially in the docs like in my website design, I don't like the idea to set the title to `pygame-ce - documentation` and would like somehow set it to just `documentation`.
> 
> What do you think ? We can just have this logo only for documentation if you want.

I totally agree with you, however, the head of the snake is currently positioned at the right edge of "pygame",
so there is no place for ce, it's not an easy process to perfectly edit the snake and move it further to the right imo.

**Comment by Mega-JC** [comment]
> Edit: I also want to add that the current pygame logo is very big and uses like a million colors. not to mention the big goofy snake and the gaming pad that it has bitten. just compare pygame's logo with some competitors: ![pyglet-1044503580](https://user-images.githubusercontent.com/122750743/244594715-5f104a81-f91c-4183-8420-ebfa987e6e81.jpeg) ![godot](https://user-images.githubusercontent.com/122750743/244594720-bbf7d601-e99f-459e-84dc-b8a88020e0cc.jpeg) ![unreal](https://user-images.githubusercontent.com/122750743/244594725-ae7556df-d04e-4c86-882c-455bd85f15b9.jpeg) ![unity](https://user-images.githubusercontent.com/122750743/244594730-00947c4d-58e8-4a5e-aea7-1d70be0a9bd2.jpeg) ![sdl](https://user-images.githubusercontent.com/122750743/244594735-18f660b5-01dc-408b-b4bc-756b695433ee.jpeg)
> 
> and now see the pygame one
> 
> ![index](https://user-images.githubusercontent.com/122750743/244594869-ae98aa03-dbf7-4f56-9e5d-ce35da1e1c99.png)
> 
> doesn't it give the impression that pygame is only made for kids? what would you think about pygame if you knew nothing about it, while looking at this logo?

> I told you people think we're a bunch of kids :/ ![adults](https://user-images.githubusercontent.com/122750743/244701879-44ad5025-8292-49b8-8a38-de4434c8abb4.png)

I have to say that this comparisons like these on the grounds of feeling "ashamed" of the artistic choices made for pygame-ce's current logo (based on a single message by an anonymous individual on an) are very unlikely to convince anyone to make logo changes.

> also, there has to be a small version/alternative for the logo that can be used as an icon. ofc there's no pygame app, but sometimes people need the compact version of a logo.

This could be useful.

**Comment by hakkeroid** [comment]
Since @Conz3D explained the intended future direction, there is probably no need to refine my suggestions according to your findings. Still, thank you for your input @yunline and @yolowex.

@Conz3D
> Like in corp. logo design, where a logo is changed only slighly in most cases of a re-design.

That's probably debatable. Many older logos from a century ago were quite fancy and detailed and got some drastic redesigns to remove their complexity and make them recognizable and memorable. Mostly logos that are already quite reduced and look "logo-like" usually have - as you said - only slight changes. I am not saying this applies to the pygame logo, just that it could be a chance. Creating a word mark with a well designed custom font is more challenging/effort than a single pictoral logo.

However, since I just discovered pygame-ce and don't know about all the details of the pygame vs ce situation, you guys know best what works for the community and the brand and what not. :+1:

**Comment by Conz3D** [comment]
> Since @Conz3D explained the intended future direction, there is probably no need to refine my suggestions according to your findings. Still, thank you for your input @yunline and @yolowex.

To be clear, I have no saying about the future direction. I'm new here, too, and I dont want to be invoved in any existing pygame drama/politics.
It was just my observation of the situation with the logo discussion and my 2ct to it.

**Comment by Mega-JC** [comment]
For those who aren't aware, I made the current iteration of the logos shown on the project READMEs of [upstream pygame](https://github.com/pygame/pygame) and pygame-ce. They are SVG versions of the inital logo design by TheCorruptor on July 29, 2001, with the "lofi" variant made by illume (someone correct me if I'm wrong) many years later.

Pygame Community's banner and other graphics are heavily based on those logos.

For what it's worth, I'm strongly against a complete design overhaul of any of pygame-ce's logos/banners and branding, especially given how recent the pygame-ce fork was created, and I believe that it would only have negative effects on the project.

Starting from scratch means discarding the 20+ years of recognizability the current logo has, and in this early stage for pygame-ce, that would massively hamper adoption, as people would be discouraged to use something that (from its logo) appears to be a "knockoff" of upstream pygame.

One thing I do agree about though is that it would be nice to have more graphics to chose from. For example, a black/white version of the logo for display in e.g. splash screens would be interesting to have, as well as a more refined version of the existing default icon used as window icons in pygame programs. All these would be put up on the [docs logo page](https://pyga.me/docs/logos.html) . A useful reference for this kind of addition is [Godot's press page](https://godotengine.org/press/).

Whether a completely new logo and branding will get created and whether more logo/icon/banner variants are made is ultimately up to the pygame-ce steering council to decide.

**Comment by hakkeroid** [comment]
@Conz3D Sorry, my bad. I misread your message as "**we** came to conclusion".

@Mega-JC thank you for the heads up.

**Comment by AndreyVarvar** [comment]
Getting rid of the original design is getting rid of the history it holds. I'm personally on the side of just modifying the current logo, no need to make a completely new one. Current logo is what everyone is used to, for example on the pygame community discord server almost all emojis/sticker revolve around the snake from the logo, and just getting rid of it makes all of that irrelevant. On the other hand, redesign would not have such a big impact on the the way people perceive pygame. After all, the pygame-ce is not a whole different library, it has deep connection with pygame itself. Shouldn't the logo represent that? I think so.

**Comment by Sjmarf** [comment]
Personally, I don't think it should be changed at all for the time being. 

Pygame CE is still relatively new - and we're trying to _replace_ the original Pygame, not be an extension of it. In a way, we _are_ the original Pygame - most of the contributors from the old repo are now working on this fork. Adding '-CE' to the logo makes us seem more like an extended version rather than a replacement. Additionally, it adds more visual clutter to a logo that's already relatively hard on the eyes.

Similarly, as Andrey said, replacing the logo entirely cuts us off from Pygame's long history and makes us seem like a cheap knock-off of upstream Pygame. I wouldn't mind tweaking the logo at _some point_ - but while Pygame CE is new, we should try to remain as true to the original as possible. 

TLDR I'm against changing it at all for the time being

**Comment by novialriptide** [comment]
> replace the original Pygame

Sure, we're trying to "replace" pygame, but shouldn't we have something to distinguish ourselves from the original pygame?

**Comment by yolowex** [comment]
> > replace the original Pygame
> 
> Sure, we're trying to "replace" pygame (although I don't think we are trying to anyway), but shouldn't we have something to distinguish ourselves from the original pygame?

that's right, everybody who contributes to pygame-ce wants the same thing. progress.
pygame-ce literally owns the community, because it has the discord server, and the subreddit. it has most of the contributors too. 
therefore there's no need to worry about pygame-ce's recognition. it has already won.

**Comment by ankith26** [comment]
We appreciate everyone's inputs on the logo (and we may as well do some sort of poll to figure out what to do about it), but just a friendly reminder that we want to keep away from any drama with upstream pygame. This fork was created with a goal of moving away from drama, and as a result this has helped (both projects) in moving on with progress in maintenance.

I'd like to make it clear that neither project "owns" the community or represents it in its entirety.

**Comment by JiffyRob** [comment]
If it's okay I would like to throw in my own two cents as well (note that it is almost entirely opinion based):
I like the pygame logo as is.  It is fun and different, just like pygame.  That said, I do understand that it can be a little noisy to look at, and is a bit too complicated to stick in the brain like some other icons do.
I do think that some change would be useful to differentiate.  I would put "community edition" below it, or something similar.  I also think that the pygame desktop icon could do with a little work.  I also think that the text could be recolored a bit to work better on more backgrounds.  It seems weird that the logo is not sitting prominently at the top of [the website](https://pyga.me/).
Other than that I wouldn't want to completely change it because we want to keep backwards compat with older games.  Say a game released two weeks ago that used the current logo as a splash screen.  If the icon was totally different, the game would be basically showcasing its outdatedness.  

I probably don't have to say this, but please don't kill Snek.

**Comment by gresm** [comment]
Regarding a splash screen, I think that it is a good idea to have some sort of a "dummy image" that comes with the library and is easily accessible. As for a real-world example backing such a decision can be Godot Engine, as with creation of the new project, it inserts its logo as one of the files and it is accessible for users in the editor to be used as a dummy image.

**Comment by Sjmarf** [comment]
> Regarding a splash screen, I think that it is a good idea to have some sort of a "dummy image" that comes with the library and is easily accessible. As for a real-world example backing such a decision can be Godot Engine, as with creation of the new project, it inserts its logo as one of the files and it is accessible for users in the editor to be used as a dummy image.

Hmmm. How would it be accessed, though? What's the syntax for loading it? Are you thinking a constant with Pygame that contains the filepath of the dummy image? DUMMY_IMAGE or smth?

**Comment by gresm** [comment]
@Sjmarf Something like that, but I'm not sure whether path is a good solution: it could generate a lot of issues with packaging and such (virtual paths, different disc read permissions on android, etc...). I was rather thinking about holding a frozen view of surface (there is an ability to lock a surface) or a binary data that could be loaded with some builtin pygame functions.

@yolowex I noticed your  :-1: , so can I ask you what exactly you don't like in that idea and why?

**Comment by gresm** [comment]
@Sjmarf I like the idea of constant (ie: ``pygame.DUMMY_IMAGE``), but I don't think that a file-path is the way to go: different operating systems have different path formats and there are enough errors with them. Packaging a pygame project can make such a file-path weird and not obvious for the library what should it be. I prefer either a binary/raw image info or a ``pygame.Surface``. The last one seems the most elegant, but having a mutable object a constant is bad. I'm not sure whether it would work, but pygame also has an option to "lock" a specified surface. If it works, in my opinion ``pygame.DUMMY_IMAGE`` (really ``pygame.locals.DUMMY_IMAGE``) should be a locked/frozen surface.

@yolowex I noticed your :-1: reaction to my previous comment. Could you elaborate on what you don't like in it and why?

**Comment by gresm** [comment]
A [thread](https://discord.com/channels/772505616680878080/1164183678923046933) on discord related to this issue was created. Here are *\*some\** of the drafts from that thread:
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/6246030d-63b3-47b2-a439-2d254e7726ff)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/e9406e6d-7ee1-4bbf-b081-bd685db3e80b)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/9647b705-5823-4627-baff-f1822e5645e3)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/998e90f7-c3e8-453f-8de9-86f153af1406)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/608b5ad8-40a9-49da-8eb3-a760fb8e1d8a)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/91ed0b9f-9e9f-482b-ab3f-ac434d10e50d)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/d5eef479-95b8-4613-a928-d470023d7eeb)
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/b12e5d8d-d0a8-4048-87ab-7179101bd633)

**Comment by Mega-JC** [comment]
Hey folks, here are the newest set of logos @kadir014 and I came up with:

Logo  |  LoFi | Powered 
:-------------------------:|:-------------------------:|:-------------------------:
![image](https://github.com/pygame-community/pygame-ce/assets/65417594/1a332d28-9dc9-49e7-911d-a999463e8baf) | ![image](https://github.com/pygame-community/pygame-ce/assets/65417594/61543307-9e67-4b85-9e3a-de8bb3e40561) | ![image](https://github.com/pygame-community/pygame-ce/assets/65417594/89d9b3b1-2cfc-405e-81e1-90f04abc00b9)

**Comment by gresm** [comment]
Great! What's left is to update the default icon, as the current one is just... 
![image](https://github.com/pygame-community/pygame-ce/assets/78505251/e8089218-33fe-4485-99a0-a03b9cb23897)

**Comment by tigercoding56** [comment]
ok  here is my take

> @hakkeroid I like your design. However, the color looks similar to the icon of Arduino ! So I tried to make another version based on yours. ![pygame](https://private-user-images.githubusercontent.com/31395137/244848571-dea0d88e-b5fd-43e9-9eb2-9e48c7689f33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzNjUyMTUsIm5iZiI6MTcyMDM2NDkxNSwicGF0aCI6Ii8zMTM5NTEzNy8yNDQ4NDg1NzEtZGVhMGQ4OGUtYjVmZC00M2U5LTllYjItOWU0OGM3Njg5ZjMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDE1MDgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzYmNmYjk5ODg3YTE3YzM3NzNmYjE2YjAwZTFhNGYxZWIwMDQxYzBlNWI0NzVkYWYyYTZhMmU4ZGFmZTdhYzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EGvH80PE_SQk5GZH4oxTqfBJiAFbJ6psC4SHdJUmuw0) I hope this would help you.

i like the more modern style, but also the classic styles , maybe we ought to have a way to select the pygame logo style you want to use in pygame like pygame.logo.set_style()

**Comment by ankith26** [comment]
I will go ahead and close this issue, as after a good amount of discussion and a poll to wrap it all up, we have decided to merge in the PR by Mega-JC which adds the logos that a majority could agree upon.

Again, thanks to everyone who contributed ideas and design works to these discussions!
